pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -121,12 +121,13 @@ func setDefaultRun(opts *SetDefaultOptions) error {
 	if opts.ViewMode {
 		if currentDefaultRepo != nil {
 			fmt.Fprintln(opts.IO.Out, displayRemoteRepoName(currentDefaultRepo))
+			return nil
 		} else if opts.IO.IsStdoutTTY() {
-			fmt.Fprintln(opts.IO.Out, "no default repository has been set; use `gh repo set-default` to select one")
+			fmt.Fprintln(opts.IO.ErrOut, "no default repository has been set; use `gh repo set-default` to select one")
+			return cmdutil.SilentError
```

Comment:
I think we'd prefer to just return an error here
```suggestion
			return fmt.Errorf("no default repository has been set; use `gh repo set-default` to select one")
```